apis/python/src/tiledbsoma/io/update_uns.py

```diff
@@ -0,0 +1,242 @@
+from os.path import join
+from typing import Literal, Optional
+
+import numpy as np
+import pandas as pd
+from somacore.options import SOMA_JOINID, PlatformConfig
+
+from tiledbsoma import DataFrame, Experiment, SOMATileDBContext
+from tiledbsoma._arrow_types import df_to_arrow
+from tiledbsoma._collection import AnyTileDBCollection, Collection
+from tiledbsoma._util import get_start_stamp
+from tiledbsoma.io._common import AdditionalMetadata, UnsMapping
+from tiledbsoma.io._registration import AxisIDMapping
+from tiledbsoma.io._registration.signatures import _prepare_df_for_ingest
+from tiledbsoma.io.ingest import (
+    IngestionParams,
+    IngestPlatformCtx,
+    _ingest_uns_array,
+    _maybe_set,
+    _write_soma_dataframe_impl,
+)
+from tiledbsoma.logging import logger
+
+Strict = Literal[True, "raise", "warn", "info", "debug", "dry_run"]
+
+
+def update_uns_uri(
+    uri: str,
+    uns: UnsMapping,
+    measurement_name: str,
+    *,
+    use_relative_uri: Optional[bool] = None,
+    context: Optional[SOMATileDBContext] = None,
+    additional_metadata: AdditionalMetadata = None,
+    platform_config: Optional[PlatformConfig] = None,
+    default_index_name: Optional[str] = None,
+    strict: Strict = True,
+) -> None:
+    """Wrapper around :func:`update_uns` that opens the experiment at the given URI.
+
+    Some "update uns" operations, that we may want to support in the future, will require opening
+    the ``Experiment`` more than once (e.g. to modify a DataFrame or array in ways that can't be
+    achieved "in-place" / at one TileDB timestamp). This function API is a sketch of what that
+    might look like, though it just calls :func:`update_uns` for now (overwriting DataFrames and
+    arrays is currently not supported, and will either ``raise`` or "INFO" log, based on the value
+    of the ``strict`` arg).
+    """
+    with Experiment.open(uri, "w") as exp:
+        update_uns(
+            exp,
+            uns,
+            measurement_name,
+            use_relative_uri=use_relative_uri,
+            context=context,
+            additional_metadata=additional_metadata,
+            platform_config=platform_config,
+            default_index_name=default_index_name,
+            strict=strict,
+        )
+
+
+def update_uns(
```

Comment:
This is going to need to be `_update_uns`, otherwise it will be public API.

Reply:
done

Reply:
Thanks!